docs/spec/sources.md

```diff
@@ -362,5 +376,37 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects).
+
+The `knsource` value SHOULD be of the form `<name>.<plural>.<group>`, where:
+
+- `name` is the [Source Custom Object](#source-custom-objects) name
+- `plural` is the [Source CRD](#source-crds)
+  [plural](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+- `group` is the [Source CRD](#source-crds)
+  [API group](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition)
+
+Examples:
+
+- `hello.pingsources.sources.knative.dev`
+- `rabbitmq-source.rabbitmqsources.sources.knative.dev`
+
+It is allowed to specify an alternative value in
+[`spec.ceOverrides.extensions`](#duckspec), in which case the `knsource` value
+can be of any shape.
+
+When the `knsource` CloudEvent extension already exists in the CloudEvent being
+forwarded by a Source, its value SHOULD be overridden as described above.
+
+One use case addressed by the `knsource` CloudEvent extension is event filtering
+based on [Source Custom Object](#source-custom-objects). For instance triggers
+can be configured to receive events coming from a single
+[Source Custom Object](#source-custom-objects).
+
```

Comment:
A lot of nits in this single sentence:
* commas
* do we need to capitalize "custom" and "object"?
* let's avoid mixing capitalization when referring to API objects (either all start with a capital letter, or none)
* I think the spec means "specific" rather than "single" here (as in "this one" versus "just one")
* that one is very subjective but the sentence sounded unfinished to me

```suggestion
based on [Source custom object](#source-custom-objects). For instance, Triggers
can be configured to receive events coming from a specific [Source custom
object](#source-custom-objects) by filtering events on this context attribute.
```

Reply:
See this thread about capitalization: https://github.com/knative/eventing/pull/3742#discussion_r462447985

Reply:
Fair enough.

Regarding the last 2 points, do I understand correctly that you prefer not to include them?

Reply:
nope, this is good. Added.